www/source/blog/2018-02-07-openshift-support.html.md

```diff
@@ -0,0 +1,40 @@
+
+For Habitat, this means that the `/hab` directory hierarchy is now readable and writable by the `root` group. This is done in such a way that a new image layer is _not_ created, meaning that container sizes are not affected.
+
+## Habitat Supervisor Changes
+
+More extensive changes were made to the Habitat Supervisor. Now, the Supervisor is aware of which Linux capabilities it needs to operate, and can adjust its behavior in the absence of those capabilities. Capabilities, for those unfamiliar with them, are a way of decomposing the all-powerful `root` user permission into a collection of largely orthogonal traits that an be individually granted to processes and files as a way of implementing the security principle of "least privilege". It turns out that to operate as a `root` user, the Supervisor only needs a small number of capabilities (`CAP_SETUID`, `CAP_SETGID`, and `CAP_CHOWN`, for the curious), which are related to spawning supervised services as a different service user, and ensuring that user can write the various state files needed by the service. When the Supervisor recognizes that it does not have these capabilities, it simply runs services as the same user that it is running as. In the context of a Docker container, this means that all services are running as the same user the container itself is running as.
+
+Additionally, the Supervisor and underlying Launcher were modified to allow processes to be run as _anonymous_ users. As stated earlier, OpenShift will run containers with large random user IDs that are all but guaranteed to not map to any named user that may exist within the container. Previously, we would take the `pkg_svc_user` and `pkg_svc_group` name that a service should run as and pass them to the Launcher, which would in turn resolve their corresponding IDs and spawn processes with them. Now, since we ignore those names and can be run as a non-`root` anonymous user, we need to resolve the IDs before they get to the Launcher, allowing us more flexibility in how to proceed.
+
+## Conclusion
+
+At the end of all this, we have a Habitat Supervisor that can successfully be run in OpenShift! By generalizing how the Supervisor behaves (and how our Docker containers are generated), we can also run in a larger variety deployment scenarios, which gives our users even more freedom in how they choose to run their Habitat applications.
```

Comment:
`larger variety deployment scenarios` -> `larger variety of deployment scenarios`